apps/balancer-tools/src/app/poolsimulator/(components)/PoolParamsForm.tsx

```diff
@@ -55,149 +45,183 @@ const inputMapper: InputMapperType = {
-    Object.entries(inputMapper).forEach(([, value]) => {
-      value.forEach((input) => {
-        setValue(input.name, undefined);
+
+    const {
+      register,
+      setValue,
+      clearErrors,
+      formState: { errors },
+      getValues,
+      trigger,
+    } = form;
+
+    useImperativeHandle(ref, () => ({
+      triggerValidation: () => {
+        trigger(); // Calls validation
+      },
+    }));
+
+    useEffect(() => {
+      clearErrors();
+      if (!poolType) return;
+
+      // Reset the form whenever poolType changes
+      Object.entries(inputMapper).forEach(([, value]) => {
+        value.forEach((input) => {
+          setValue(input.name, undefined);
+        });
       });
-    });
-    setInitialData({
-      poolParams: undefined,
-      tokens: [],
-      poolType: poolType,
-    });
-  }, [poolType]);
-
-  useEffect(() => {
-    register("tokens", { required: true, value: initialData?.tokens });
-  }, []);
-  return (
-    <Form {...form} onSubmit={onSubmit} id="initial-data-form">
-      <div className="flex flex-col gap-3">
-        {inputMapper[poolType].map((input) => (
-          <div className="relative">
-            <FormField
-              name={input.name}
-              render={({ field }) => (
-                <Input
-                  {...field}
-                  label={input.label}
-                  type="number"
-                  validation={{
-                    required: true,
-                    valueAsNumber: true,
-                    value: initialData.poolParams?.[input.name],
-                  }}
-                  defaultValue={initialData.poolParams?.[input.name]}
-                  placeholder={input.placeholder}
-                />
-              )}
-            />
-            <span className="absolute top-8 right-2 flex items-center text-slate10">
-              {input.unit}
-            </span>
-          </div>
-        ))}
-        <div className="flex flex-col">
-          <label className="mb-2 block text-sm text-slate12">Tokens</label>
-          {errors?.tokens?.message && (
-            <div className="mt-1 h-6 text-sm text-tomato10">
-              <span>{errors?.tokens?.message as string}</span>
+
+      inputMapper[poolType].forEach((input) => {
+        const dataValue = data.poolParams?.[input.name];
+        if (dataValue) {
+          setValue(input.name, input.transformFromDataToForm(dataValue));
+        }
+      });
+      if (data?.tokens) setValue("tokens", data?.tokens);
+    }, [data.poolParams, data.tokens, poolType]);
+
+    useEffect(() => {
+      register("tokens", { required: true, value: data?.tokens });
+    }, []);
+
+    const { value: currentTab } = useTabContext();
+
+    useEffect(() => {
+      const data = getValues();
+      onTabChanged(createPayload(poolType, data));
+    }, [currentTab]);
+
+    // useEffect(()=> {
+    //   if (errors.length) {setChangeTabAllowed(false)} else {setChangeTabAllowed(true)}
+    // },[errors])
```

Comment:
This is a TODO or just a leftover from the previous version? I would add a comment if it's a todo or delete this